### PR TITLE
Make IndexingMemoryController private to IndicesService

### PR DESCRIPTION
**Pull request by s1monw**
This commit detaches the IndexingMemoryController from guice and moves
it's creation and closing into IndicesService where it logically belongs.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java [hunk 1]
```diff
@@ -131,29 +124,23 @@ protected IndexingMemoryController(Settings settings, ThreadPool threadPool, Ind
...
-    protected void doStart() {
+    protected ScheduledFuture<?> scheduleTask(ThreadPool threadPool) {
         // it's fine to run it on the scheduler thread, no busy work
-        this.scheduler = threadPool.scheduleWithFixedDelay(statusChecker, interval);
+        return threadPool.scheduleWithFixedDelay(statusChecker, interval);
     }
 
-    @Override
-    protected void doStop() {
+    public void close() {
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java]: Shouldn't we have an @Override here?

**Comment by mikemccand** [comment]
LGTM

**Comment by jasontedor** [comment]
Yay! LGTM.

**Comment by bleskes** [comment]
late to the game , but +1 :)

**Pull request merged**